src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionEmitCachingMemberAccessor.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NETFRAMEWORK || NETCOREAPP
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+namespace System.Text.Json.Serialization.Metadata
+{
+    internal sealed partial class ReflectionEmitCachingMemberAccessor : MemberAccessor
+    {
+        private static readonly ReflectionEmitMemberAccessor s_sourceAccessor = new();
+        private static readonly Cache<(string id, Type declaringType, MemberInfo? member)> s_cache =
+            new(slidingExpiration: TimeSpan.FromSeconds(5), evictionInterval: TimeSpan.FromSeconds(1));
```

Comment:
> Could these values be hardcoded in the ctor since this is the only ctor reference?

What's the difference? I think it's better if these parameters are surfaced in the main implementation file.

> Also, how did you land on these values?

Good question. The purpose of the cache is to promote dynamic methods to long-living allocations that we want to eventually evict if unused (and since the cache is unbounded we don't need to evict on insertion). At the same time the eviction routine can be relatively expensive in the pathological scenaria we are looking to optimize (traversing through thousands of constantly reused member accessors) so it makes sense to spread it out as much as possible. This suggests to me that that both expiration and eviction intervals should be in the order of hundreds of milliseconds to a few seconds. The actual constants were chosen arbitrarily, although as might be expected choices doesn't demonstrate substantial variation in the context of microbenchmarks. We can always adjust these values in the future.